src/cc/frontends/clang/loader.cc

```diff
@@ -155,23 +153,58 @@ int ClangLoader::parse(unique_ptr<llvm::Module> *mod, TableStorage &ts,
     return -1;
   if (flags_ & DEBUG_SOURCE)
     flags_cstr.push_back("-g");
-  kflags.push_back("-include");
-  kflags.push_back("/virtual/include/bcc/bpf.h");
-  kflags.push_back("-include");
-  kflags.push_back("/virtual/include/bcc/helpers.h");
-  kflags.push_back("-isystem");
-  kflags.push_back("/virtual/include");
   for (auto it = kflags.begin(); it != kflags.end(); ++it)
     flags_cstr.push_back(it->c_str());
+
+  vector<const char *> flags_cstr_rem;
+  flags_cstr_rem.push_back("-include");
+  flags_cstr_rem.push_back("/virtual/include/bcc/helpers.h");
+  flags_cstr_rem.push_back("-isystem");
+  flags_cstr_rem.push_back("/virtual/include");
   if (cflags) {
     for (auto i = 0; i < ncflags; ++i)
-      flags_cstr.push_back(cflags[i]);
+      flags_cstr_rem.push_back(cflags[i]);
   }
 #ifdef CUR_CPU_IDENTIFIER
   string cur_cpu_flag = string("-DCUR_CPU_IDENTIFIER=") + CUR_CPU_IDENTIFIER;
-  flags_cstr.push_back(cur_cpu_flag.c_str());
+  flags_cstr_rem.push_back(cur_cpu_flag.c_str());
 #endif
 
+  // SharedTableStorage is global per process and ts may contain tables
+  // from previous modules
+  if (do_compile(mod, ts, in_memory, flags_cstr, flags_cstr_rem, main_path,
+                 main_buf, id, func_src, mod_src, true)) {
+    // try one more time to compile with system bpf.h
+    llvm::errs() << "WARNING: compilation failure, trying use system bpf.h\n";
+
+    func_src.clear();
+    mod_src.clear();
```

Comment:
Hmm, this may have other repercussions that may need to be rolled back or stashed temporarily. Would it be possible to get a compile error after the frontend action has run, resulting in some tables being opened twice?

Reply:
Just checked it is possible that fd is created twice for a table. Will try to roll back before the second compile.

I actually thought this is no harm earlier since we may just waste a fd. But think twice, this gets reference counted in kernel and may not be a good idea.